CARET_trace/src/ros_trace_points.cpp

```diff
@@ -51,6 +52,92 @@
 std::unique_ptr<std::thread> trace_node_thread;
 thread_local bool trace_filter_is_rcl_publish_recorded;
 
+using std::string;
+using std::vector;
+static bool ignore_rcl_timer_init = false;
+
+static vector<string> string_split(string & str, char delim)
+{
+  using std::stringstream;
+  vector<string> elems;
+  stringstream ss(str);
+  string item;
+  while (getline(ss, item, delim)) {
+    if (!item.empty()) {
+      elems.push_back(item);
+    }
+  }
+  return elems;
+}
+
+static bool is_ros2_launch_command()
+{
+  using std::ifstream;
+  using std::ios;
+  using std::istreambuf_iterator;
+  using std::to_string;
+  pid_t pid = getpid();
+  string fileName = "/proc/" + to_string(pid) + "/cmdline";
+
+  ifstream ifs(fileName, ios::in | ios::binary);
+  if (!ifs) {
+    return false;
+  }
+  istreambuf_iterator<char> it_ifs_begin(ifs);
+  istreambuf_iterator<char> it_ifs_end{};
+  vector<char> input_data(it_ifs_begin, it_ifs_end);
+  if (!ifs) {
+    ifs.close();
+    return false;
+  }
+  ifs.close();
+
+  vector<string> cmd_line;
+  auto itr_begin = input_data.begin();
+  auto itr_find = input_data.begin();
+  for (;;) {
+    itr_find = find(itr_begin, input_data.end(), 0x00);
+    if (itr_find == input_data.end()) {
+      break;
+    }
+    string output_string(itr_begin, itr_find);
+    cmd_line.push_back(output_string);
+    itr_begin = itr_find + 1;
+  }
+
+  // Checkpoint 1 : cmdline[0] contains "python"
+  if (cmd_line.size() <= 2) {
+    return false;
+  }
+  if (cmd_line[0].find("python") == string::npos) {
+    return false;
+  }
+
+  // Checkpoint 2 : Matches "ros2" to the right of the last "/" in cmdline[1]
+  if (cmd_line[1].find("ros2") == string::npos) {
+    return false;
+  }
+  auto subStr = string_split(cmd_line[1], '/');
+  if (subStr[subStr.size() - 1].compare("ros2") != 0) {
+    return false;
+  }
+
+  // Checkpoint 3 : The first argument after cmdline[2] that does not start with "-" matches
```

Comment:
Could you tell me when cmdline[2] start with "-" ?

Reply:
Although it is not related to this autoware command,
In the ISP internal review, if there is an option specification in the "/opt/ros/humble/bin/ros2" command, that is, if there is a "-" before "launch", it is allowed

Reply:
Understood!

Reply:
@miyakoshi-dev 
Could you modify the comment to explain it?
e.g. 
The first argument after cmdline[2] excluding options (starting with "-") matches

Reply:
Updated with comments as I reviewed the conditions